core/src/main/java/kafka/server/share/ShareFetchUtils.java

```diff
@@ -128,4 +130,17 @@ static long offsetForEarliestTimestamp(TopicIdPartition topicIdPartition, Replic
                 Optional.empty(), true).timestampAndOffsetOpt();
```

Comment:
Unfortunately, I think this is more complicated that it seems. If KIP-405 is enabled, then returning `EARLIEST_TIMESTAMP` is asynchronous. It may be necessary to wait before there's an answer to the earliest offset. If we put that concern aside for this PR, we do need an issue to track this.